### PR TITLE
Ui

### PR DESCRIPTION
**Pull request by amitu**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -18,3 +18,6 @@ test-runner/target
 venv
 target-nix
 __pycache__
+
+.zcompdump-ACKO354-5.7.1
+.zcompdump-ip-192-168-43-14-5.7.1
```
Comment (amitu) [inline .gitignore]: Make this `.zcomdump-*`.

File: fbt_lib/src/lib.rs [hunk 2]
```diff
@@ -1,13 +1,22 @@
+use crate::types::{Failure, SingleTestResult, TestResult};
 use std::path::PathBuf;
+use std::time::{Duration, Instant};
```
Comment (amitu) [inline fbt_lib/src/lib.rs]: Avoid `use` statements. Put entire path. You can make the path slightly better, eg put a `pub use types::*` in `lib.rs`, so you can do `pub fn test_all() -> anyhow::Result<crate::TestResult> {`.

File: fbt_lib/src/lib.rs [hunk 3]
```diff
@@ -1,13 +1,22 @@
...
 
 pub mod types;
 
-pub fn test_all() -> anyhow::Result<bool> {
+pub fn test_all() -> anyhow::Result<TestResult> {
```
Comment (amitu) [inline fbt_lib/src/lib.rs]: Create custom error type using `thiserror` library. Refer [this blog post](https://nick.groenen.me/posts/rust-error-handling/).

File: src/main.rs [hunk 4]
```diff
@@ -1,7 +1,46 @@
...
+                            id.blue().to_string(),
+                            "FAILED".red().to_string(),
+                            format!("{:?}", &test.duration).yellow().to_string()
+                        );
+                    }
+                }
+            }
+            // println!("Test Results {:#?}", res)
+        }
+        Err(e) => eprintln!("failed: {:?}", e),
```
Comment (amitu) [inline src/main.rs]:
```suggestion
    let r = match fbt_lib::test_all() {
        Ok(r) => r,
        Err(e) => return eprintln!("failed: {:?}", e),
    };
    
    let mut single_results: Vec<SingleTestResult> = vec![];
    match test_result.results {
        Ok(result) => single_results = result,
        _ => {}
    }
    for test in single_results.iter() {
        let id = test.id.clone();
        match &test.result {
            Ok(status) => {
                if *status {
                    println!(
                        "Test: {}, Status: {}, Time: {}",
                        id.blue().to_string(),
                        "SUCCESS".green().to_string(),
                        format!("{:?}", &test.duration).yellow().to_string()
                    );
                } else {
                    println!(
                        "Test: {}, Status: {}, Time: {}",
                        id.blue().to_string(),
                        "SKIPPED".magenta().to_string(),
                        format!("{:?}", &test.duration).yellow().to_string()
                    );
                }
            }
            Err(_e) => {
                println!(
                    "Test: {}, Status: {}, Time: {}",
                    id.blue().to_string(),
                    "FAILED".red().to_string(),
                    format!("{:?}", &test.duration).yellow().to_string()
                );
            }
        }
    }  
```

Comment (amitu) [reply src/main.rs]: Above will "de-indent" the code. You should try that as much.

File: .gitignore (hunk 1 above)
Comment (amitu) [inline .gitignore]:
```suggestion

.zcompdump-*
```

File: fbt_lib/src/lib.rs [hunk 5]
```diff
@@ -1,20 +1,27 @@
-use std::path::PathBuf;
-
 pub mod types;
 
-pub fn test_all() -> anyhow::Result<bool> {
+use std::time::{Duration, Instant};
```
Comment (amitu) [inline fbt_lib/src/lib.rs:3]: Remove this. Dont use `use` statements as much as possible.

Comment (spattk) [reply fbt_lib/src/lib.rs:3]: will keep in mind.

**Pull request merged**